Inherit gRPC client cluster validation configuration from a common configuration (#10883)
grafana/mimir: CHANGELOG.md

  * `-server.cluster-validation.label`
  * `-server.cluster-validation.grpc.enabled`
  * `-server.cluster-validation.grpc.soft-validation`

> * [ENHANCEMENT] gRPC clients: Add experimental support to include the cluster validation label in gRPC metadata. When cluster validation is enabled on gRPC server side, the cluster validation label from gRPC metadata is compared with gRPC server's cluster validation label. #10869 #10883

Reviewer: ```suggestion
* [ENHANCEMENT] gRPC clients: Add experimental support to include the cluster validation label in gRPC metadata. When cluster validation is enabled on gRPC server side, the cluster validation label from gRPC metadata is compared with the gRPC server's cluster validation label. #10869 #10883
```

>   * By setting `-<grpc-client-config-path>.cluster-validation.label`, we configure the cluster validation label of _a single_ gRPC client, whose `grpcclient.Config` object is configurable via `-<grpc-client-config-path>`.

Reviewer: ```suggestion
  * By setting `-<grpc-client-config-path>.cluster-validation.label`, you configure the _a single_ gRPC client cluster validation label. You can configure this label's `grpcclient.Config` object through `-<grpc-client-config-path>`.
```

>   * By setting `-common.client-cluster-validation.label`, we configure the cluster validation label of _all_ gRPC clients.

Reviewer: ```suggestion
  * By setting `-common.client-cluster-validation.label`, you configure the _all_ gRPC clients cluster validation label.```

> * [ENHANCEMENT] gRPC clients: Add `cortex_client_request_invalid_cluster_validation_labels_total` metrics, that is used by Mimir's gRPC clients to track invalid cluster validations. #10767

Reviewer: ```suggestion
* [ENHANCEMENT] gRPC clients: Add `cortex_client_request_invalid_cluster_validation_labels_total` metrics, that are used by Mimir's gRPC clients to track invalid cluster validations. #10767
```
